test/PrintAsObjC/cdecl-official.swift

```diff
@@ -0,0 +1,85 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t --leading-lines
+
+/// Generate cdecl.h
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) \
+// RUN:   %t/Lib.swift -emit-module -verify -o %t -emit-module-doc \
+// RUN:   -emit-objc-header-path %t/cdecl.h \
+// RUN:   -enable-experimental-feature CDecl
+
+/// Check cdecl.h directly
+// RUN: %FileCheck %s --input-file %t/cdecl.h
+// RUN: %check-in-clang %t/cdecl.h
+// RUN: %check-in-clang-c %t/cdecl.h -Wnullable-to-nonnull-conversion
+// RUN: %check-in-clang-cxx %t/cdecl.h
+
+/// Build a client against cdecl.h
+// RUN: %clang -c %t/Client.c -fmodules -I %t \
+// RUN:   -F %S/../Inputs/clang-importer-sdk-path/frameworks \
+// RUN:   -I %clang-include-dir -Werror \
+// RUN:   -isysroot %S/../Inputs/clang-importer-sdk
+
+// REQUIRES: swift_feature_CDecl
+
+//--- Lib.swift
+
+// CHECK: #if defined(__cplusplus)
```

Comment:
Strongly recommend adding test cases that require an import and then testing the imports you get. 

Remember, to be compatible with non-clang compilers, official cdecl needs to *not* use `@import`; it needs to use `#include` instead (or at least fall back to it). It also needs to not use the `.h`-less `#include`s that are common in C++.

Reply:
Yeah, C includes with require some new logic to the printing component.